packages/read-model/src/Product/Listed/ListedProductViewElasticFacade.php

```diff
@@ -0,0 +1,276 @@
+     */
+    protected $domain;
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Model\Customer\CurrentCustomer
+     */
+    protected $currentCustomer;
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Model\Product\TopProduct\TopProductFacade
+     */
+    protected $topProductFacade;
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Model\Product\ProductOnCurrentDomainFacadeInterface
+     */
+    protected $productOnCurrentDomainFacade;
+
+    /**
+     * @var \Shopsys\ReadModelBundle\Product\Listed\ListedProductViewFactory
+     */
+    protected $listedProductViewFactory;
+
+    /**
+     * @var \Shopsys\ReadModelBundle\Image\ImageViewFacade
+     */
+    protected $imageViewFacade;
+
+    /**
+     * @var \Shopsys\ReadModelBundle\Product\Action\ProductActionViewFacade
+     */
+    protected $productActionViewFacade;
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Product\ProductFacade $productFacade
+     * @param \Shopsys\FrameworkBundle\Model\Product\Accessory\ProductAccessoryFacade $productAccessoryFacade
+     * @param \Shopsys\FrameworkBundle\Component\Domain\Domain $domain
+     * @param \Shopsys\FrameworkBundle\Model\Customer\CurrentCustomer $currentCustomer
+     * @param \Shopsys\FrameworkBundle\Model\Product\TopProduct\TopProductFacade $topProductFacade
+     * @param \Shopsys\FrameworkBundle\Model\Product\ProductOnCurrentDomainFacadeInterface $productOnCurrentDomainFacade
+     * @param \Shopsys\ReadModelBundle\Product\Listed\ListedProductViewFactory $listedProductViewFactory
+     * @param \Shopsys\ReadModelBundle\Product\Action\ProductActionViewFacade $productActionViewFacade
+     * @param \Shopsys\ReadModelBundle\Image\ImageViewFacade $imageViewFacade
+     */
+    public function __construct(
+        ProductFacade $productFacade,
+        ProductAccessoryFacade $productAccessoryFacade,
+        Domain $domain,
+        CurrentCustomer $currentCustomer,
+        TopProductFacade $topProductFacade,
+        ProductOnCurrentDomainFacadeInterface $productOnCurrentDomainFacade,
+        ListedProductViewFactory $listedProductViewFactory,
+        ProductActionViewFacade $productActionViewFacade,
+        ImageViewFacade $imageViewFacade
+    ) {
+        $this->productFacade = $productFacade;
+        $this->productAccessoryFacade = $productAccessoryFacade;
+        $this->domain = $domain;
+        $this->currentCustomer = $currentCustomer;
+        $this->topProductFacade = $topProductFacade;
+        $this->productOnCurrentDomainFacade = $productOnCurrentDomainFacade;
+        $this->listedProductViewFactory = $listedProductViewFactory;
+        $this->productActionViewFacade = $productActionViewFacade;
+        $this->imageViewFacade = $imageViewFacade;
+    }
+
+    /**
+     * @param int $limit
+     * @return \Shopsys\ReadModelBundle\Product\Listed\ListedProductView[]
+     */
+    public function getTop(int $limit): array
+    {
+        $topProducts = $this->topProductFacade->getAllOfferedProducts(
```

Comment:
This "Elastic" implementation still hydrates a lot of entities. Is it intentional? Couldn't it be a reason for bad performance gain?

Reply:
It is intentional. I don't really think that it could be that slow because of it but maybe we could create user story or issue to put these into Elasticsearch as well